examples/binaries/waiter/tests/rw_lock_access.rs

```diff
@@ -0,0 +1,182 @@
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::Drop),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn as_ref_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::AsRef),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn as_mut_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::AsMut),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn deref_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::Deref),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn deref_mut_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::DerefMut),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+fn init_fixture(system: &System, lock_type: RwLockType) -> (Program<'_>, MessageId) {
+    system.init_logger_with_default_filter("");
```

Comment:
why?

Reply:
For the testing purposes I build the contract with the `debug` feature on which enables log messages. It has turned out, by default `system.init_logger()` enables all debug messages, but I am not keen on seeing them in tests